### PR TITLE
Support for persistent volumes, alternate namespaces, and changing external CIDR

### PR DESCRIPTION
**Pull request by mluds**
Persistent volumes: I added PVCs for postgres and the keep store. It now seems to keep everything intact after restarting.

Alternate namespaces: It was using the "default" namespace by default, which caused DNS issues when using a different namespace. I changed it to use the namespace from Helm.

Changing External CIDR: For me the nginx proxy was trying to route external requests to the internal keep nodes, causing them to fail. I made the external CIDR configurable to fix this.

### CODE REVIEW

File: charts/arvados/values.yaml [hunk 1]
```diff
@@ -77,6 +77,20 @@ arvados:
       arvadosCLI: 2.1.0
       arvadosLoginSync: 2.1.0
 
+postgres:
+  persistence:
+    enabled: true
+    # Leave this null to use the default class
+    storageClass: ~
+    # The rancher.io/local-path storage class does not support a size limit
```
Comment (cure) [inline charts/arvados/values.yaml]: This comment seems like it was accidentally included from a copy-paste, do we need it?

File: charts/arvados/templates/keep-store-deployment.yaml [hunk 2]
```diff
@@ -33,13 +33,28 @@ spec:
           command:
             - "sh"
             - "-c"
-            - "mkdir /keepdata && /usr/local/bin/bootstrap.sh keepstore={{ .Values.arvados.versions.distribution.keepStore }} && GOGC=10 keepstore"
+            - "mkdir -p /keepdata && /usr/local/bin/bootstrap.sh keepstore={{ .Values.arvados.versions.distribution.keepStore }} && GOGC=10 keepstore"
```
Comment (cure) [inline charts/arvados/templates/keep-store-deployment.yaml]: The -p seems superfluous? The directory being created is only one level deep.

**Review by cure — COMMENTED** [review]
Thanks! I have a few small nits below that should be easy to fix.

I tried the persistent volumes. They work as expected for the Keep volumes. The Postgresql volume, however, does not persist when deleting the arvados chart with helm. The difference is down to the fact that Keep is implemented as a StatefulSet, and Postgresql is not. Perhaps we should switch Posgresql to a StatefulSet (with replicas: 1). Does that sound OK?

**Comment by cure** [comment]
Here's a patch that converts postgresql to a StatefulSet:
```
diff --git a/charts/arvados/templates/postgres-deployment.yaml b/charts/arvados/templates/postgres-deployment.yaml
index 8462bb6..d820e8e 100644
--- a/charts/arvados/templates/postgres-deployment.yaml
+++ b/charts/arvados/templates/postgres-deployment.yaml
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
 metadata:
   name: "arvados-postgres"
   labels:
@@ -12,6 +12,9 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
+  replicas: 1
+  podManagementPolicy: Parallel
+  serviceName: arvados-postgres
   selector:
     matchLabels:
       app: arvados-postgres
@@ -22,6 +25,7 @@ spec:
         app: arvados-postgres
         release: {{ .Release.Name }}
     spec:
+      terminationGracePeriodSeconds: 1
       containers:
         - name: {{ .Chart.Name }}
           image: "postgres:9.5"
@@ -46,3 +50,15 @@ spec:
           persistentVolumeClaim:
             claimName: {{ .Release.Name }}-postgres-pvc
         {{- end }}
+  {{- if .Values.postgres.persistence.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: postgres-storage
+      spec:
+        storageClassName: {{ .Values.postgres.persistence.storageClass }}
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: {{ .Values.postgres.persistence.size }}
+  {{- end }}
diff --git a/charts/arvados/templates/postgres-pvc.yaml b/charts/arvados/templates/postgres-pvc.yaml
deleted file mode 100644
index 85bcbc2..0000000
--- a/charts/arvados/templates/postgres-pvc.yaml
+++ /dev/null
@@ -1,22 +0,0 @@
-# Copyright (C) The Arvados Authors. All rights reserved.
-#
-# SPDX-License-Identifier: Apache-2.0
-
-{{- if .Values.postgres.persistence.enabled }}
-apiVersion: v1
-kind: PersistentVolumeClaim
-metadata:
-  name: {{ .Release.Name }}-postgres-pvc
-  labels:
-    app: {{ template "arvados.name" . }}
-    chart: {{ template "arvados.chart" . }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
-spec:
-  storageClassName: {{ .Values.postgres.persistence.storageClass }}
-  accessModes:
-    - ReadWriteOnce
-  resources:
-    requests:
-      storage: {{ .Values.postgres.persistence.size }}
-{{- end }}
\ No newline at end of file
```
It seems to work for me. Let me know if you'd like to merge your commit with the nits fixed, and with this applied on top?

File: charts/arvados/values.yaml (hunk 1 above)
Comment (mluds) [reply charts/arvados/values.yaml]: Probably not. It's a little unrelated so I removed it.

File: charts/arvados/templates/keep-store-deployment.yaml (hunk 2 above)
Comment (mluds) [reply charts/arvados/templates/keep-store-deployment.yaml]: Thanks, I fixed this

**Comment by mluds** [comment]
Thanks, I fixed the nits and added your patch in a new commit. If you wanted to add this, just let me know and I'll remove it.

**Comment by cure** [comment]
Thank you! It's merged.

**Pull request merged**